lib/publiccloud/instance.pm

```diff
@@ -85,6 +86,43 @@ sub run_ssh_command {
     }
 }
 
+=head2 ssh_script_retry
+
+    ssh_script_retry(command[, retry => 3][, delay => 10][, timeout => 90][, ssh_opts =>'..'][, username => 'XXX'][, no_quote => 0]);
+
+Run a C<command> via ssh in the given PC instance until it succeeds or
+the given number of retries is exhausted and an exception is thrown.
+Timeout can be set by C<timeout> or 90 sec by default.
+By default, the command is passed in single quotes to SSH.
+To avoid quoting use C<<no_quote=>1>>.
+With C<<ssh_opts=>'...'>> you can overwrite all default ops which are in
+C<<$instance->ssh_opts>>.
+Use argument C<username> to specify a different username then
+C<<$instance->username()>>.
+=cut
+sub ssh_script_retry {
+    my ($self, $cmd, %args) = @_;
+    $cmd = $args{cmd} if (!defined $cmd && $args{cmd});
+    $args{rc_only} = 1;
+    my $tries = $args{retry} // 3;
+    my $delay = $args{delay} // 10;
+
+    for (my $try = 0; $try < $tries; $try++) {
+        my $rc = 1;
+        # ssh_script_run can throw a timeout exception
+        eval {
+            $rc = $self->run_ssh_command($cmd, %args);
+        };
+        if ($@ or $rc != 0) {
+            sleep($delay);
+            next;
+        } else {
+            return $rc;
+        }
+    }
+    die "ssh_script_retry failed for command '$cmd'";
```

Comment:
I'd wait for Godot instead!
```suggestion
    die "Waiting for Godot: ".$cmd;
```

Reply:
https://en.wikipedia.org/wiki/Waiting_for_Godot ? :P

Reply:
Of course :)

Reply:
Done! :+1: